woocommerce/woocommerce-ios
# [Order Fulfillment][Part 2] Update Review Order screen with Products section

Part 2 of #4136
For more context: [part 1](https://github.com/woocommerce/woocommerce-ios/pull/4534), [part 3](https://github.com/woocommerce/woocommerce-ios/pull/4539), [part 4](https://github.com/woocommerce/woocommerce-ios/pull/4549), [part 5](https://github.com/woocommerce/woocommerce-ios/pull/4567).

# Description
This PR follows up #4534 (please makes sure to review that PR first) and updates Review Order screen with Product section.

# Solution
- ReviewOrderViewModel was updated with new subtypes `Section` and `Row` to define sections and rows of the review order table view.
- The view model acts as data source for the review order table view, so it was updated to config displayed data for the Product section including header title and cell view model.
- Product cell view models require info about add-ons, so the view model was updated to receive `showAddOns` config from Order Details screen. Add-ons are also queried to check if they're available on the products.
- ReviewOrderViewController was updated to config its table view's datasource and delegate. Action to show add-ons was also implemented.

# Screenshot
<img src="https://user-images.githubusercontent.com/5533851/124424597-76ea6000-dd91-11eb-8736-c70db4a2defe.gif" width=400 />

# Testing
- Navigate to Orders tab
- Select an order with In Progress status
- Select Mark Order Complete
- You should then be navigated to Review Order screen with only one section named Products.
- If `showAddOns` config is enabled, the displayed product cells should show option to see add-ons appropriately (like in the gif above).

Update release notes:
